pkg/controller/serverside/applier.go

```diff
@@ -0,0 +1,40 @@
+package serverside
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/eks-anywhere/pkg/clients/kubernetes"
+	"github.com/aws/eks-anywhere/pkg/controller"
+	"github.com/aws/eks-anywhere/pkg/controller/clientutil"
+)
+
+type ObjectGenerator func() ([]kubernetes.Object, error)
+
+type ObjectApplier struct {
+	client client.Client
+}
+
+func NewObjectApplier(client client.Client) *ObjectApplier {
+	return &ObjectApplier{
+		client: client,
+	}
+}
+
+func (a *ObjectApplier) Apply(ctx context.Context, generateObjects ObjectGenerator) (controller.Result, error) {
```

Comment:
Why are we passing in a func instead of just passing in kubernetes.Object as an array or opts?

Reply:
That's just for convenience. If I passed the objects, I wouldn't need this applier, I could just use the `serverside.ReconcileObjecst` method.

This allows me to pass the generator of objects (method implemented in the snow package) and let this applier do the error handling.

Reply:
Yea I guess that's what I was confused about, I feel it makes it a bit harder to understand the flow because when I think of `Apply` I think of passing in the objects rather than a func that generates them. Does it make sense to call this `ApplyWithGenerator`, however I am assuming that having "with" in naming is probably not ideal.

Reply:
I see I see. Given that the only purpose of this object is to server as a helper for this, isn't it just redundant info?

What about renaming the struct to `ObjectGenerateApplier` or `ApplierForGenerator`?